core/src/test/scala/tech/beshu/ror/integration/ActionYamlLoadedAccessControlTest.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.integration
+
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.Inside
+import org.scalatest.matchers.should.Matchers._
+import org.scalatest.wordspec.AnyWordSpec
+import tech.beshu.ror.accesscontrol.AccessControl.RegularRequestResult.{Allow, ForbiddenByMismatched}
+import tech.beshu.ror.accesscontrol.domain.Action
+import tech.beshu.ror.mocks.MockRequestContext
+
+class ActionYamlLoadedAccessControlTest extends AnyWordSpec with BaseYamlLoadedAccessControlTest with Inside {
+
+  override protected def configYaml: String =
+    """
+      |readonlyrest:
+      |
+      |  access_control_rules:
+      |
+      |  - name: "Allowed for user metadata action"
+      |    type: "allow"
+      |    actions: ["cluster:internal_ror/user_metadata/get"]
+      |
+      |  - name: "Allowed for test config action"
+      |    type: "allow"
+      |    actions: ["cluster:internal_ror/testconfig/*"]
+      |
+      |  - name: "Allowed for auth mock action"
+      |    type: "allow"
+      |    actions: ["cluster:ror/authmock/manage"] # old format
+      |
+      |  - name: "Allowed for config action"
+      |    type: "allow"
+      |    actions: ["cluster:ror/config/*"] # old format
+      |
+      |  - name: "Allowed for any action"
+      |    type: "allow"
+      |    actions: ["cluster:r*"] # old format
+    """.stripMargin
+
+  "An ACL" when {
+    "actions rule is defined" should {
+      "allow to proceed" when {
+        "it is an metadata request and the action is on the configured list with new name" in {
+          val request = MockRequestContext.metadata.copy(action = Action.RorAction.RorUserMetadataAction)
+          val result = acl.handleRegularRequest(request).runSyncUnsafe()
+          result.history should have size 1
```

Comment:
maybe instead of this assertion, let's assert matched block name (it's more readable when test fails)

Reply:
Allow is defined like that `final case class Allow[B <: BlockContext](blockContext: B, block: Block)` ;) we don't have to have `lastMatchedBlockName` method